internal/build/custom_builder_test.go

```diff
@@ -37,13 +39,13 @@ func TestCustomBuildSuccessClusterRefTaggedWithDigest(t *testing.T) {
 	defer f.teardown()
 
 	sha := digest.Digest("sha256:11cd0eb38bc3ceb958ffb2f9bd70be3fb317ce7d255c8a4c3f4af30e298aa1aab")
-	f.dCli.Images["localhost:1234/foo/bar:tilt-build-1551202573"] = types.ImageInspect{ID: string(sha)}
+	f.dCli.Images["localhost:1234/foo_bar:tilt-build-1551202573"] = types.ImageInspect{ID: string(sha)}
```

Comment:
this change isn't right.

I think you broke something here, and it's revealing a deeper problem with this PR?

The ReplaceRegistry function you're using isn't bijective, and it's leading to problems where going from
imageName -> (registry, path) -> imageName
is making a different image name pop out.

The ReplaceRegistry function you're using isn't bijective -> i.e.,

Reply:
sorry! you were right, i misread the test